### PR TITLE
Fix w3c validation warning type style

### PR DESCRIPTION
**Pull request by mdugue**
# Summary
Removes `type="text/css"` from the `<style />` tag, as it causes the warnings 

> The `type` attribute for the `style` element is not needed and should be omitted.

in w3c validator, see https://validator.w3.org/nu/?doc=https%3A%2F%2Fwww.styled-components.com%2F for an example.

Motivated by https://spectrum.chat/?thread=085d48f2-47e2-43ac-8459-500722a0ba3f

Tests looked fine, snapshots changed as expected. I couldn't see any problems with _SSR_, _jest-styled-components_ or _babel-plugin-styled-components_.

## TBD
Does styled-components support *html 4* environments? If so, this PR should be adjusted to add the `type` attribute only if in html5-"mode"

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,414 +2,416 @@
 
 All notable changes to this project will be documented in this file. If a contribution does not have a mention next to it, [@geelen](https://github.com/geelen) or [@mxstbr](https://github.com/mxstbr) did it.
 
-*The format is based on [Keep a Changelog](http://keepachangelog.com/) and this project adheres to [Semantic Versioning](http://semver.org/).*
```
Comment (kitten) [inline CHANGELOG.md]: Can you remove your formatting from this file please? It’s be great to preserve its previous history and format

**Comment by probablyup** [comment]
It used to be before html5. Now it’s just a legacy thing.

File: CHANGELOG.md (hunk 1 above)
Comment (mdugue) [reply CHANGELOG.md]: sorry, didn't realise 🤔

**Comment by mdugue** [comment]
@kitten right, its not hurting to include the `type` attribute, but since HTML5 it default to `text/css` and w3c validation considers its declaration redundant. 

Do we have to consider _pre html5_ environments?

File: CHANGELOG.md (hunk 1 above)
Comment (mdugue) [reply CHANGELOG.md]: fixed

**Comment by kitten** [comment]
I’m a little concerned about IE9, but if this  works for it that’d be awesome 😎

**Comment by mdugue** [comment]
I tested the sandbox in IE8, which didn't seem to have any Problems 😃

**Comment by mdugue** [comment]
Btw. I just checked https://emotion.sh/ out of curiosity. It doesn't seem to set `type` attributes either

**Comment by mxstbr** [comment]
Thank you so much for helping us improve styled-components! Based on our [Community Guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md) every person that has a PR of any kind merged is offered **an invitation to the styled-components organization**—that is you right now!

Should you accept, you'll get write access to the main repository. (and a fancy `styled-components` logo on your GitHub profile!) You'll be able to label and close issues, merge pull requests, create new branches, etc. We want you to help us out with those things, so don't be scared to do them! Make sure to read our [contribution and community guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md), which explains all of this in more detail. Of course, if you have any questions just let me know!

**Pull request merged**